core/src/main/java/org/apache/calcite/sql/SqlIntervalQualifier.java

```diff
@@ -1131,6 +1149,111 @@ private int[] evaluateIntervalLiteralAsSecond(
     }
   }
 
+  /**
+   * Validates an INTERVAL literal against an MILLISECOND interval qualifier.
+   *
+   * @throws org.apache.calcite.runtime.CalciteContextException if the interval
+   * value is illegal
+   */
+  private int[] evaluateIntervalLiteralAsMillisecond(
+      RelDataTypeSystem typeSystem, int sign,
+      String value,
+      String originalValue,
+      SqlParserPos pos) {
+    BigDecimal millisecond;
+
+    // validate as MINUTE(startPrecision), e.g. 'MM'
+    String intervalPattern = "(\\d+)";
+
+    Matcher m = Pattern.compile(intervalPattern).matcher(value);
+    if (m.matches()) {
+      // Break out  field values
+      try {
+        millisecond = parseField(m, 1);
```

Comment:
I'm a bit confused, this mentions validating it as `MINUTE(startPrecision)`, but here and the rest of the function you're operating on the value as if it's a milisecond. Can you update this comment and/or add clarification?

Reply:
updated comment